app/controllers/npq_separation/admin/milestones_controller.rb

```diff
@@ -0,0 +1,92 @@
+class NpqSeparation::Admin::MilestonesController < NpqSeparation::AdminController
+  before_action :set_cohort
+  before_action :set_schedule
+  before_action :ensure_super_admin
+  before_action :set_milestone, only: %i[edit update destroy]
+  before_action :set_new_statement_date, only: %i[create update]
+  before_action :set_statement_date_options, only: %i[new edit]
+
+  def new
+    @declaration_types_available = Milestone::DECLARATION_TYPES - @schedule.milestones.pluck(:declaration_type)
+  end
+
+  def create
+    ActiveRecord::Base.transaction do
+      milestone = @schedule.milestones.create!(params.fetch(:milestone).permit(:declaration_type))
+      milestone.milestones_statements.find_or_create_by!(
+        statement: Statement.find_by!(cohort: @cohort, year: @new_statement_date.year, month: @new_statement_date.month),
+      )
```

Comment:
This is only creating MilestoneStatements for the first LP it encounters so this then wont be available for inclusion on Statements for other LPs?

Unless I'm missing something I think this needs to be creating a `milestones_statements` record for each LP? 

Considering its all in a transaction I'm wondering whether this should using a service object. That service object could have `save` and `destroy` actions which would then 'sync' the milestone statements with the attributes the service object holds?